hs/src/Reach/Verify/SMT.hs

```diff
@@ -311,6 +317,130 @@ seVars se =
 set_to_seq :: S.Set a -> Seq.Seq a
 set_to_seq = Seq.fromList . S.toList
 
+type BindingEnv = M.Map String (Maybe DLVar, SrcLoc, BindingOrigin, Maybe SExpr, Maybe DLExpr)
+
+-- Log every occurence of dlvars so we know what is used how many times
+dlvOccurs :: [Int] -> BindingEnv -> DLExpr -> [Int]
```

Comment:
This function might be what `Counts.hs` is about

Reply:
After looking at it, I don't think it's appropriate. After counting a variable, I need to expand it into the value it's associated with in `bindings` and process that value as well.

Reply:
Ok!